praktikumsplaner-frontend/frontend/src/views/Praktikumsplaetze/MeldungAusbildung.vue

```diff
@@ -416,40 +418,69 @@ const customMenuProps = {
 };
 const form = ref<HTMLFormElement>();
 const formatter = useFormatter();
-const meldezeitraeume = ref<object[]>([]);
-const isAusbidungsleitung = ref<boolean>(false);
+const meldezeitraeume = computed(() => {
+    return [
+        currentMeldezeitraum.value,
+        ...upcomingMeldezeitraeume.value,
+        ...passedMeldezeitraeume.value,
+    ];
+});
+const currentMeldezeitraum = ref<Meldezeitraum>();
+const upcomingMeldezeitraeume = ref<Meldezeitraum[]>([]);
+const passedMeldezeitraeume = ref<Meldezeitraum[]>([]);
 
 onMounted(() => {
     MeldezeitraumService.getCurrentMeldezeitraum()
         .then((zeitraueme) => {
             activeMeldezeitraum.value = zeitraueme.length > 0;
+            currentMeldezeitraum.value = zeitraueme[0];
+            currentMeldezeitraum.value.zeitraumName = formatZeitraum(
+                currentMeldezeitraum.value
+            );
         })
         .then(() => {
             if (
                 userStore.getRoles.includes("ROLE_AUSBILDUNGSLEITUNG") ||
                 APP_SECURITY !== "true"
             ) {
-                isAusbidungsleitung.value = true;
+                isAusbildungsleitung.value = true;
                 activeMeldezeitraum.value = true;
-                MeldezeitraumService.getAllMeldezeitraeume().then(
-                    (zeitraeume) => {
-                        meldezeitraeume.value = zeitraeume.map((zeitraum) => {
-                            return {
-                                id: zeitraum.id,
-                                zeitraumName: `${
-                                    zeitraum.zeitraumName
-                                }: ${formatter.formatDateFromString(
-                                    zeitraum.zeitraum.startZeitpunkt
-                                )} - ${formatter.formatDateFromString(
-                                    zeitraum.zeitraum.endZeitpunkt
-                                )}`,
-                            };
-                        });
-                    }
-                );
+                getUpcomingMeldezeitraeume();
+                getPassedMeldezeitraeume();
             }
         });
 });
+
+function getUpcomingMeldezeitraeume() {
+    MeldezeitraumService.getUpcomingMeldezeitraueme().then((zeitraeume) => {
+        upcomingMeldezeitraeume.value = zeitraeume.map((zeitraum) => {
+            return {
+                ...zeitraum,
+                zeitraumName: formatZeitraum(zeitraum),
+            };
+        });
+    });
+}
+
+function getPassedMeldezeitraeume() {
+    MeldezeitraumService.getPassedMeldezeitraueme().then((zeitraeume) => {
+        passedMeldezeitraeume.value = zeitraeume.map((zeitraum) => {
+            return {
+                ...zeitraum,
+                zeitraumName: formatZeitraum(zeitraum),
+            };
+        });
+    });
+}
```

Comment:
Maybe we could use the slots provided by v-select to customize the formatting of item-text and use the Meldezeitraum data without mapping it like this.

---

 `<template slot="selection" slot-scope="data">
    <!-- HTML that describe how select should render selected items -->
    {{ data.item.name }} - {{ data.item.description }}
  </template>
  <template slot="item" slot-scope="data">
    <!-- HTML that describe how select should render items when the select is open -->
    {{ data.item.name }} - {{ data.item.description }}
  </template>`